src/controllers/Document.ts

```diff
@@ -773,6 +773,43 @@ export class DocumentController extends BaseController {
       .then(response => response.result);
   }
 
+  /**
+   * Applies a partial update to an existing document.
+   * If the document doesn't already exist, a new document is created.
+   * @see https://docs.kuzzle.io/sdk/js/7/controllers/document/upsert/
+   *
+   * @param index Index name
+   * @param collection Collection name
+   * @param _id Unique document identifier
+   * @param changes Partial content of the document to update
+   * @param options
```

Comment:
(nitpicking) optional arguments are to be put between brackets

```suggestion
   * @param [options]
```